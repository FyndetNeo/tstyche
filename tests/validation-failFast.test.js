import { afterEach, describe, expect, test } from "@jest/globals";
import { clearFixture, getFixtureUrl, writeFixture } from "./__utils__/fixtureFactory.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const fixtureUrl = getFixtureUrl("validation-failFast", { generated: true });

afterEach(async () => {
  await clearFixture(fixtureUrl);
});

describe("'failFast' configuration file option", () => {
  test("when specified value is not boolean", async () => {
    const config = {
      failFast: "never",
    };

    await writeFixture(fixtureUrl, {
      ["tstyche.config.json"]: JSON.stringify(config, null, 2),
    });

    const { exitCode, stderr, stdout } = await spawnTyche(fixtureUrl);

    expect(stdout).toBe("");
    expect(stderr).toMatchSnapshot("stderr");

    expect(exitCode).toBe(1);
  });
});
