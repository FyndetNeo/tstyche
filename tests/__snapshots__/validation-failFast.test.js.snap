// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'failFast' configuration file option when specified value is not boolean: stderr 1`] = `
"Error: Option 'failFast' requires an argument of type boolean.

  1 | {
> 2 |   "failFast": "never"
    |               ^
  3 | }

      at ./tstyche.config.json:2:15

"
`;
